src/AVSGateway.js

```diff
@@ -0,0 +1,141 @@
+import { urls, paths } from "./AVSEndPoints";
+import { cannedErrorResponses, customErrorCodes } from "./CannedErrorResponses";
+import { cannedResponses } from "./CannedResponses";
+
+import IllegalArgumentError from "./errors/IllegalArgumentError";
+
+const uuid = require("uuid/v4");
+const util = require("util");
+const { hasIn } = require("immutable");
+
+const parser = require("./SpeakDirectiveParser");
+
+// TODO: Once the region setting is made configurable by the user, the URLS
+// need to ge generated as against hard coded.
+const EVENTS_URL = urls.NA + paths.EVENTS;
+
+// Default error repsonse if we encounter any unexpected errors.
+const ERROR_MESSAGE = cannedErrorResponses.get(customErrorCodes.UNKNOWN_ERROR);
+
+/**
+ * Manages interactions with AVS
+ */
+class AVSGateway {
+  /**
+   * Sends the TextMessage event to AVS and extracts Alexa's response.
+   *
+   * @param {String} userRequestToAlexa The multi-part response from AVS. This
+   * should not be empty or undefined.
+   * @param {String} accessToken The multi-part response from AVS. This
+   * should not be empty or undefined.
+   *
+   * @returns The text response from Alexa. An empty response is possible if Alexa
+   * said nothing. If an error happens while communicating to AVS or while parsing
+   * the responses, a canned human-readable error message is returned.
+   *
+   * @throws IllegalArgumentError if the input is missing or invalid.
+   */
+  async sendTextMessageEvent(userRequestToAlexa, accessToken) {
+    if (!userRequestToAlexa || !accessToken) {
+      throw new IllegalArgumentError(
+        `The request string to Alexa or the access token cannot be empty.
+        RequestString: ${userRequestToAlexa}
+        AccessToken: ${accessToken}`
+      );
+    }
+
+    const testMessageEvent = JSON.stringify(
+      this.buildTextMessageEvent(userRequestToAlexa)
+    );
+    const requestOptions = this.buildTextMessageFetchRequestOptions(
+      testMessageEvent,
+      accessToken
+    );
+
+    let isOk = false;
+    let payload;
+    let responseFromAlexa = ERROR_MESSAGE;
```

Comment:
Defensive coding. Its like setting fields like isLoggedIn to false by default. If in future we screw it up somehow and end up not populating it, we will at least render an error message to the user (which is justified because we screwed up something) instead of indeterministic behavior resulting from an uninitialized field.